src/main.rs

```diff
@@ -29,50 +28,38 @@ fn main() {
             }),
             synchronous_pipeline_compilation: false,
         }),))
+        //States
+        .insert_state(AppState::Game)
+        //State transitions
         // Debug plugins.
         .add_plugins(InspectorPlugin)
-        // Physics plugins.
+        //Our plugins
         .add_plugins((
-            RapierPhysicsPlugin::<NoUserData>::default(),
-            RapierDebugRenderPlugin::default(),
+            GamePlugin,
+            MainMenuPlugin
         ))
         // Third party plugins.
         .add_plugins(LdtkPlugin)
-        // Castle Fight plugins.
-        .add_plugins((
-            CastleFightLdtkPlugin,
-            ResourcesPlugin,
-            CameraPlugin,
-            WaypointPlugin,
-            VisionPlugin,
-            AttackPlugin,
-            HealthPlugin,
-            MovementPlugin,
-            BuildingSpawningPlugin,
-            UnitSpawningPlugin,
-        ))
         // Third party resources
         .insert_resource(LevelSelection::index(0))
```

Comment:
This is setting up our map loaded from the LDtk file. It could perhaps be moved to the game folder?